src/platform/plugins/shared/data_views/common/data_views/data_views.ts

```diff
@@ -1320,6 +1320,8 @@ export class DataViewsService {
       .then((response) => {
         indexPattern.id = response.id;
         indexPattern.version = response.version;
+        this.dataViewCache.set(indexPattern.id, Promise.resolve(indexPattern as DataView));
+        this.dataViewLazyCache.set(indexPattern.id, Promise.resolve(indexPattern as DataViewLazy));
```

Comment:
if we wouldn't  update the cache, we would need to clean the cache if users changed the data view. updating the cache is much more efficient, then cleaning/loading it all.